widgets.py

```diff
@@ -475,4 +475,5 @@ def loadFilterSymbol(self):
     def zoomToFilter(self):
         if self.controller.hasValidFilter():
             iface.mapCanvas().setExtent(self.controller.currentFilter.geometry.boundingBox())
```

Comment:
Another option would be to use [zoomToFeatureExtent](https://qgis.org/pyqgis/master/gui/QgsMapCanvas.html#qgis.gui.QgsMapCanvas.zoomToFeatureExtent) instead of `zoomByFactor`:

```suggestion
            iface.mapCanvas().zoomToFeatureExtent(self.controller.currentFilter.geometry.boundingBox())
```